libs/datamodel/core/src/transform/ast_to_dml/lift.rs

```diff
@@ -194,24 +195,111 @@ impl<'a> LiftAstToDml<'a> {
     ) -> Result<(dml::FieldType, Vec<ast::Directive>), DatamodelError> {
         let type_name = &ast_field.field_type.name;
 
+        let (supports_native_types, datasource_name) = match self.source {
+            Some(source) => (source.has_preview_feature("nativeTypes"), source.name.as_str()),
+            _ => (false, ""),
+        };
+
         if let Ok(scalar_type) = ScalarType::from_str(type_name) {
-            if USE_CONNECTORS_FOR_CUSTOM_TYPES {
-                let pg_connector = SqlDatamodelConnectors::postgres();
-                let pg_type_specification = ast_field.directives.iter().find(|dir| dir.name.name.starts_with("pg.")); // we use find because there should be at max 1.
-                let name = pg_type_specification.map(|dir| dir.name.name.trim_start_matches("pg."));
-                let args = pg_type_specification
-                    .map(|dir| {
-                        let args = dir
-                            .arguments
-                            .iter()
-                            .map(|arg| ValueValidator::new(&arg.value).as_int().unwrap() as u32)
-                            .collect();
-                        args
-                    })
-                    .unwrap_or(vec![]);
-
-                if let Some(x) = name.and_then(|ts| pg_connector.parse_native_type(&ts, args)) {
-                    let field_type = dml::FieldType::NativeType(scalar_type, x);
+            if supports_native_types {
+                let (connector_string, connector) = match self.source {
+                    Some(source) => (&source.active_provider, &source.active_connector),
+                    None => panic!(""),
+                };
+                let mut connectors: HashMap<&str, Box<dyn Connector>> = HashMap::new();
+                connectors.insert("postgresql", Box::new(SqlDatamodelConnectors::postgres()));
+                connectors.insert("mysql", Box::new(SqlDatamodelConnectors::mysql()));
+                connectors.insert("sqlite", Box::new(SqlDatamodelConnectors::sqlite()));
+                connectors.insert("mssql", Box::new(SqlDatamodelConnectors::mssql()));
+
+                let prefix = format!("{}{}", datasource_name, ".");
+
+                let type_specifications = ast_field
+                    .directives
+                    .iter()
+                    .filter(|dir| dir.name.name.starts_with(&prefix))
+                    .collect_vec();
+
+                if type_specifications.len() > 1 {
+                    return Err(DatamodelError::new_duplicate_directive_error(
+                        &prefix,
+                        type_specifications.first().unwrap().span,
+                    ));
+                }
```

Comment:
Might be good to introduce a variable `type_specification` so that you don't have to call `type_specifications.first()` everywhere.